tests/models/encoder_decoders/wikitables/wikitables_semantic_parser_test.py

```diff
@@ -0,0 +1,310 @@
+        assert embedded_actions.size(0) == 4
+        assert action_indices[(0, 0)] == action_indices[(1, 0)]
+        assert len(set(action_indices.values())) == 4
+
+        # Now we'll go through all four unique actions and make sure the embedding is as we expect.
+        action_embedding = embedded_actions[action_indices[(0, 0)]]
+        left_side_embedding = nonterminal_embedding(rule2_tensor)
+        right_side_embedding = terminal_encoder(char2_tensor.unsqueeze(0).unsqueeze(0)).squeeze(0)
+        expected_action_embedding = torch.cat([left_side_embedding, right_side_embedding],
+                                              dim=-1).squeeze(0)
+        assert_almost_equal(action_embedding.cpu().data.numpy(),
+                            expected_action_embedding.cpu().data.numpy())
+
+        action_embedding = embedded_actions[action_indices[(0, 1)]]
+        left_side_embedding = nonterminal_embedding(rule3_tensor)
+        right_side_embedding = terminal_encoder(char3_tensor.unsqueeze(0).unsqueeze(0)).squeeze(0)
+        expected_action_embedding = torch.cat([left_side_embedding, right_side_embedding],
+                                              dim=-1).squeeze(0)
+        assert_almost_equal(action_embedding.cpu().data.numpy(),
+                            expected_action_embedding.cpu().data.numpy())
+
+        action_embedding = embedded_actions[action_indices[(1, 1)]]
+        left_side_embedding = nonterminal_embedding(rule4_tensor)
+        right_side_embedding = terminal_encoder(char4_tensor.unsqueeze(0).unsqueeze(0)).squeeze(0)
+        expected_action_embedding = torch.cat([left_side_embedding, right_side_embedding],
+                                              dim=-1).squeeze(0)
+        assert_almost_equal(action_embedding.cpu().data.numpy(),
+                            expected_action_embedding.cpu().data.numpy())
+
+        action_embedding = embedded_actions[action_indices[(1, 2)]]
+        left_side_embedding = nonterminal_embedding(start_tensor)
+        right_side_embedding = nonterminal_embedding(rule2_tensor)
+        expected_action_embedding = torch.cat([left_side_embedding, right_side_embedding],
+                                              dim=-1).squeeze(0)
+        assert_almost_equal(action_embedding.cpu().data.numpy(),
+                            expected_action_embedding.cpu().data.numpy())
+
+        # Finally, we'll check that the embedding for the initial action is as we expect.
+        start_embedding = nonterminal_embedding(start_tensor).squeeze(0)
+        zeros = Variable(start_embedding.data.new(start_embedding.size(-1)).fill_(0).float())
+        expected_action_embedding = torch.cat([zeros, start_embedding], dim=-1)
+        assert_almost_equal(initial_action_embedding.cpu().data.numpy(),
+                            expected_action_embedding.cpu().data.numpy())
+
+    def test_embed_actions_does_not_embed_terminals_when_set(self):
```

Comment:
You mean `test_embed_actions_does_not_embed_terminals_when_not_set`?

Reply:
This test is also removed in the next PR.